packages/kit/src/components/Connect.tsx

```diff
@@ -14,23 +18,95 @@ export const Connect: React.FC<ConnectProps> = ({
   onDisconnect,
 }) => {
   const {user, authenticate, unauthenticate} = useCurrentFlowUser()
+  const [open, setOpen] = useState(false)
+  const [copied, setCopied] = useState(false)
 
-  const handleConnect = async () => {
-    await authenticate()
-    onConnect?.()
+  const displayAddress =
+    user?.loggedIn && user.addr
+      ? `${user.addr.slice(0, 6)}...${user.addr.slice(-4)}`
+      : ""
+
+  const handleButtonClick = async () => {
+    if (user?.loggedIn) {
+      setOpen(true)
+    } else {
+      await authenticate()
+      onConnect?.()
+    }
+  }
+
+  const handleCopy = async () => {
+    if (
+      user?.addr &&
+      typeof window !== "undefined" &&
+      window.navigator.clipboard
+    ) {
+      await window.navigator.clipboard.writeText(user.addr)
+      setCopied(true)
+      setTimeout(() => setCopied(false), 1500)
+    }
   }
 
   const handleDisconnect = () => {
     unauthenticate()
+    setOpen(false)
     onDisconnect?.()
   }
 
   return (
-    <Button
-      variant={variant}
-      onClick={user?.loggedIn ? handleDisconnect : handleConnect}
-    >
-      {user?.loggedIn ? "Disconnect Wallet" : "Connect Wallet"}
-    </Button>
+    <>
+      <Button
+        onClick={handleButtonClick}
+        variant={user?.loggedIn ? "outline" : variant}
+        className="px-2"
+      >
+        {user?.loggedIn ? displayAddress : "Connect Wallet"}
+      </Button>
+      {user?.loggedIn && (
+        <Dialog isOpen={open} onClose={() => setOpen(false)}>
+          <div className="flex flex-col items-center gap-4 min-w-[320px]">
+            <div className="flex flex-col items-center">
+              <div className="w-16 h-16 rounded-full bg-slate-100 flex items-center justify-center mb-2">
+                <CircleUserRoundIcon className="w-8 h-8 text-slate-400" />
+              </div>
+              <div className="text-center text-lg font-semibold mb-0">
+                {displayAddress}
+              </div>
+              <div className="text-center text-sm text-gray-500 -mt-1">
+                1,416.00 FLOW
```

Comment:
Is this a typo or a placeholder for flow balance?

Reply:
Yep there's another ticket to add balance